### PR TITLE
[ENH] Use TSV instead of CSV for tabular files

### LINKED ISSUE #247: Move from CSV files to TSV
To be more consistent with Neurobagel/BIDS. This is an easy change but requires us to manually convert files in existing datasets

Files to update:
- Manifest
- Doughnut
- Bagel

Other
- Update dataframe-saving function to write TSV instead of CSV
- Update tabular data-loading method to load TSV instead of CSV

### PR DESCRIPTION
**Pull request by michellewang**
- Closes #247

Changes proposed in this pull request:
- Convert all CSV files to TSV
- Read/write tabular data as TSV instead of CSV
- Update layout
- Update docs

## Checklist (for reviewers)
_This section is for the PR reviewer_

- [x] PR has an interpretable title with a prefix (e.g. `[BUG]`, `[DOC]`, `[ENH]`, `[MAINT]`)\
Refer to [NumPy Development Guide](https://numpy.org/doc/stable/dev/development_workflow.html#writing-the-commit-message) for a full list
- [x] PR links to GitHub issue with mention `Closes #XXXX`
- [ ] Tests pass
- [ ] Checks pass

For new features:
- [ ] Tests have been added

For bug fixes:
- [ ] There is at least one test that would fail under the original bug conditions

### CODE REVIEW

File: nipoppy/tabular/base.py [hunk 1]
```diff
@@ -91,6 +91,8 @@ def load(cls, fpath: StrOrPathLike, validate=True, **kwargs) -> Self:
                 "This function does not accept 'dtype' as a keyword argument"
                 ". Everything is read as a string and (optionally) validated later."
             )
+        if "sep" not in kwargs:
+            kwargs["sep"] = "\t"
         df = cls(pd.read_csv(fpath, dtype=str, **kwargs))
```
Comment (michellewang) [inline nipoppy/tabular/base.py:96]: Consider checking that the TSV is actually a TSV and not a CSV: https://github.com/neurobagel/bagel-cli/blob/5d15125cdce8c14357d159e3b377c0b0ec7ac8b1/bagel/pheno_utils.py#L44-L84

**Review by surchs — APPROVED** [review]
Looks good to me. On a quick browse the only place I saw the "write tab" case was in the "save with backup" in BaseTabular. Does that mean you rely on the default being tsv now to control that you're making tsvs?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -17,6 +17,7 @@ repos:
     -   id: check-yaml
     -   id: end-of-file-fixer
     -   id: trailing-whitespace
+        exclude: .*\.tsv
```
[the remark this answers is not in the document]
Comment (michellewang) [reply .pre-commit-config.yaml:20]: If the last column has missing values then those lines just end with a tab which is stripped if we don’t ignore TSVs for this check. But I’ll check again to see if we still have cases like that and if not can remove this

[the remark this answers is not in the document]
Comment (michellewang) [reply .pre-commit-config.yaml:20]: Update: can confirm that this is still a possible case in the manifest, which allows empty values for the `datatype` column.
- https://github.com/nipoppy/nipoppy/blob/af05f4515776941546cf305e01949b6abba2c884/nipoppy/data/examples/sample_manifest.tsv
- https://github.com/nipoppy/nipoppy/blob/af05f4515776941546cf305e01949b6abba2c884/tests/data/manifest2.tsv
- https://github.com/nipoppy/nipoppy/blob/af05f4515776941546cf305e01949b6abba2c884/tests/data/manifest3.tsv

**Comment by michellewang** [comment]
Thanks for the review @surchs! I made it more explicit that we were using tabs as separators now in `BaseTabular`

**Pull request merged**